CMakeLists.txt

```diff
@@ -10,6 +10,7 @@ cmake_dependent_option(ES_GLES "Build the game with OpenGL ES" OFF UNIX OFF)
 cmake_dependent_option(ES_STEAM "Build the game for the Steam Linux runtime" OFF UNIX OFF)
 cmake_dependent_option(ES_USE_SYSTEM_LIBRARIES "Use system libraries instead of the vcpkg ones." ON "APPLE OR ES_STEAM" OFF)
 cmake_dependent_option(ES_CREATE_BUNDLE "Create a Bundle instead of an executable. Not suitable for development purposes." OFF APPLE OFF)
+cmake_dependent_option(ES_PARALLEL_USE_TASK_QUEUE "Use the TaskQueue backend for parallel operations." OFF "UNIX" ON)
```

Comment:
I prefer not to have options like this. Just choose the (overall) best implementation and only support that one.
It does make sense to have a switch like this if we want to temporary support parallel-enabled and parallel-disabled versions of the game (as part of a risk mitigation), but we are not doing that in this PR, so that doesn't seem relevant.

Reply:
It's used for the vcpkg issue I mentioned in another comment, and having the option available on all OSes makes testing easier.

I suppose this would be removed alongside most of Parallel.h once the vcpkg port is fixed.